examples/seismic/viscoacoustic/operators.py

```diff
@@ -193,4 +497,57 @@ def ForwardOperator(model, geometry, space_order=4, kernel='blanch_symes', save=
                     name='Forward', **kwargs)
 
 
-kernels = {'blanch_symes': blanch_symes, 'ren': ren, 'deng_mcmechan': deng_mcmechan}
+def AdjointOperator(model, geometry, space_order=4, kernel='sls', time_order=2, **kwargs):
+    """
+    Construct an adjoint modelling operator in a viscoacoustic medium.
+
+    Parameters
+    ----------
+    model : Model
+        Object containing the physical parameters.
+    geometry : AcquisitionGeometry
+        Geometry object that contains the source (SparseTimeFunction) and
+        receivers (SparseTimeFunction) and their position.
+    space_order : int, optional
+        Space discretization order.
+    kernel : selects a visco-acoustic equation from the options below:
+        sls (Standard Linear Solid) :
+        1st order - Blanch and Symes (1995) / Dutta and Schuster (2014)
+        viscoacoustic equation
+        2nd order - Bai et al. (2014) viscoacoustic equation
+        ren - Ren et al. (2014) viscoacoustic equation
+        deng_mcmechan - Deng and McMechan (2007) viscoacoustic equation
+        Defaults to sls 2nd order.
+    """
+    dt = model.grid.time_dim.spacing
+    m = model.m
+
+    u = VectorTimeFunction(name="u", grid=model.grid, time_order=time_order,
+                           space_order=space_order)
+    q = TimeFunction(name="q", grid=model.grid, save=None, time_order=time_order,
+                     space_order=space_order, staggered=NODE)
+    srca = PointSource(name='srca', grid=model.grid, time_range=geometry.time_axis,
+                       npoint=geometry.nsrc)
+    rec = Receiver(name='rec', grid=model.grid, time_range=geometry.time_axis,
+                   npoint=geometry.nrec)
+
+    # Equations kernels
+    eq_kernel = kernels[kernel]
+    eqn = eq_kernel(model, geometry, u, q, forward=False)
+
+    # Construct expression to inject receiver values
+    receivers = rec.inject(field=q.backward, expr=rec * dt**2 / m)
```

Comment:
2nd order acoustic weight `dt**2 / m` pde dependent.

Reply:
Okay. We will fix it.